nix/shell.nix

```diff
@@ -98,7 +98,7 @@
             name = "cardano-cli";
             help = "CLI v9.2.1 that is used in partner-chains dependency stack";
```

Comment:
hey @tgunnoe , can you please upgrade the dev shell to use the latest cardano node 10.1.2. We'll also need to update the protocolVersion on it's shelley/genesis.json to 10. This will put us in the Chang+1 HF.

Thanks!